drivers/kw2xrf/Kconfig

```diff
@@ -16,9 +16,6 @@ menuconfig MODULE_KW2XRF
     select MODULE_CORE_THREAD_FLAGS
     select MODULE_IEEE802154
     select MODULE_LUID
-    select MODULE_NETDEV
-    select MODULE_NETDEV_IEEE802154
-    select MODULE_NETDEV_LEGACY_API
```

Comment:
Since those should already be provided by the submac netdev, the removal here is correct either way.